clar/sandbox.h

```diff
@@ -75,6 +71,34 @@ find_tmp_path(char *buffer, size_t length)
 	return -1;
 }
 
+static int canonicalize_tmp_path(char *buffer)
+{
+#ifdef _WIN32
+	char tmp[CLAR_PATH_MAX];
+	DWORD ret;
+
+	ret = GetFullPathName(buffer, CLAR_PATH_MAX, tmp, NULL);
+
+	if (ret == 0 || ret > CLAR_PATH_MAX)
+		return -1;
+
+	ret = GetLongPathName(tmp, buffer, CLAR_PATH_MAX);
+
+	if (ret == 0 || ret > CLAR_PATH_MAX)
+		return -1;
+
+	return 0;
+#else
+	char tmp[CLAR_PATH_MAX];
+
+	if (realpath(buffer, tmp) == NULL)
```

Comment:
I worry a bit that realpath may not be available on all systems, as it is only specified by POSIX.1-2001 and thus not part of C90 itself. And unfortunately we have folks that compile Git on systems that predate POSIX.1-2001 itself.

I sometimes wish that some kind of resource existed that provides a matrix of operating system <-> availability of certain features.